PR: https://github.com/odoo/odoo/pull/

From: e38e2515882dacc73ab7d87fa76da3c5a395c6a6
From: Mahendra barad
Date: 2017-09-06 13:21:25

Breaking data model changes scores: del:17 + add:0, change matches:
-    module_website_form_editor = fields.Boolean("Custom Forms")
-    module_website_twitter = fields.Boolean("Twitter Roller")
-    module_website_blog = fields.Boolean("Blogs")
-    module_website_livechat = fields.Boolean("Live Chat")
-    module_website_forum = fields.Boolean("Forum")
-    module_website_crm = fields.Boolean("Contact Form")
-    module_website_slides = fields.Boolean("Slides")
-    module_website_hr_recruitment = fields.Boolean("Jobs")
-    module_website_sale = fields.Boolean("eCommerce")
-    module_sale_subscription = fields.Boolean("Subscriptions")
-    module_website_event_sale = fields.Boolean("Event Tickets")
-    social_twitter = fields.Char('Twitter Account')
-    social_facebook = fields.Char('Facebook Account')
-    social_github = fields.Char('GitHub Account')
-    social_linkedin = fields.Char('LinkedIn Account')
-    social_youtube = fields.Char('Youtube Account')
-    social_googleplus = fields.Char('Google+ Account')

Total Changes: 43

[IMP] website: Clean and simplify res_config view

Purpose
=======

* Settings are too complex: too long + too many options that shouldn't be suggested with checkboxes

e.g. twitter roller is now suggested as a new snippet to install from website editor
google maps, slides, forum, etc. should show up in the apps store

* When saving the page I expect to stay in the settings -> no redirection to homepage anymore!

Specification
=============

See task 33620

================================= pseudo patch: =================================

--- a/addons/website/models/res_company.py
+++ b/addons/website/models/res_company.py
@@ -1,13 +1,20 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models
+from odoo import api, fields, models
 
 
 class Company(models.Model):
-
     _inherit = "res.company"
 
+    # NB: Also defined in mass_mailing
+    social_twitter = fields.Char('Twitter Account')
+    social_facebook = fields.Char('Facebook Account')
+    social_github = fields.Char('GitHub Account')
+    social_linkedin = fields.Char('LinkedIn Account')
+    social_youtube = fields.Char('Youtube Account')
+    social_googleplus = fields.Char('Google+ Account')
+
     @api.multi
     def google_map_img(self, zoom=8, width=298, height=298):
         partner = self.sudo().partner_id

--- a/addons/website/models/res_config_settings.py
+++ b/addons/website/models/res_config_settings.py
@@ -21,29 +21,10 @@ class ResConfigSettings(models.TransientModel):
     google_management_client_id = fields.Char('Google Client ID', related='website_id.google_management_client_id')
     google_management_client_secret = fields.Char('Google Client Secret', related='website_id.google_management_client_secret')
 
-    social_twitter = fields.Char("Twitter", related='website_id.social_twitter')
-    social_facebook = fields.Char("Facebook", related='website_id.social_facebook')
-    social_github = fields.Char("GitHub", related='website_id.social_github')
-    social_linkedin = fields.Char("LinkedIn", related='website_id.social_linkedin')
-    social_youtube = fields.Char("Youtube", related='website_id.social_youtube')
-    social_googleplus = fields.Char("Google+", related='website_id.social_googleplus')
-
     cdn_activated = fields.Boolean('Use a Content Delivery Network (CDN)', related='website_id.cdn_activated')
     cdn_url = fields.Char(related='website_id.cdn_url')
     cdn_filters = fields.Text(related='website_id.cdn_filters')
-
-    module_website_form_editor = fields.Boolean("Custom Forms")
     module_website_version = fields.Boolean("A/B Testing")
-    module_website_twitter = fields.Boolean("Twitter Roller")
-    module_website_blog = fields.Boolean("Blogs")
-    module_website_livechat = fields.Boolean("Live Chat")
-    module_website_forum = fields.Boolean("Forum")
-    module_website_crm = fields.Boolean("Contact Form")
-    module_website_slides = fields.Boolean("Slides")
-    module_website_hr_recruitment = fields.Boolean("Jobs")
-    module_website_sale = fields.Boolean("eCommerce")
-    module_sale_subscription = fields.Boolean("Subscriptions")
-    module_website_event_sale = fields.Boolean("Event Tickets")
 
     favicon = fields.Binary('Favicon', related='website_id.favicon')
     # Set as global config parameter since methods using it are not website-aware. To be changed

--- a/addons/website/models/website.py
+++ b/addons/website/models/website.py
@@ -49,12 +49,13 @@ class Website(models.Model):
     default_lang_id = fields.Many2one('res.lang', string="Default Language", default=_default_language, required=True)
     default_lang_code = fields.Char(related='default_lang_id.code', string="Default language code", store=True)
 
-    social_twitter = fields.Char('Twitter Account')
-    social_facebook = fields.Char('Facebook Account')
-    social_github = fields.Char('GitHub Account')
-    social_linkedin = fields.Char('LinkedIn Account')
-    social_youtube = fields.Char('Youtube Account')
-    social_googleplus = fields.Char('Google+ Account')
+    social_twitter = fields.Char(related="company_id.social_twitter")
+    social_facebook = fields.Char(related="company_id.social_facebook")
+    social_github = fields.Char(related="company_id.social_github")
+    social_linkedin = fields.Char(related="company_id.social_linkedin")
+    social_youtube = fields.Char(related="company_id.social_youtube")
+    social_googleplus = fields.Char(related="company_id.social_googleplus")
+
     google_analytics_key = fields.Char('Google Analytics Key')
     google_management_client_id = fields.Char('Google Client ID')
     google_management_client_secret = fields.Char('Google Client Secret')
